### ISSUE TITLE
Include release date

### ISSUE THREAD

**Issue by gusutabopb** [zanieb/rooster#47]
(Coming here from `uv`)

As an avid `uv` CHANGELOG reader, I think it'd be nice to have the release date next to the version number, as recommended by [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).

Current output
```
## 0.4.2
```

Better (IMHO) output
```
## 0.4.2 - 2024-09-01
```

**Comment by zanieb** [comment]
Hi! Sorry for the delay, my notifications here are buried by all the uv ones I guess :)

Would you be willing to have the release date after the heading?

**Comment by gusutabopb** [comment]
Yes, that format is fine too. That’s similar to what CPython uses: https://docs.python.org/3/whatsnew/changelog.html
